server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -1690,6 +1691,21 @@ public boolean isRecovering() {
      */
     public abstract void maybePruneDeletes();
 
+    /**
+     * Returns the maximum auto_id_timestamp of all append-only have been processed (or force-updated) by this engine.
+     * Notes this method returns the auto_id_timestamp of all append-only requests, not max_unsafe_auto_id_timestamp.
+     * @see #forceUpdateMaxUnsafeAutoIdTimestamp(long)
+     */
+    public long getMaxSeenAutoIdTimestamp() {
+        return IndexRequest.UNSET_AUTO_GENERATED_TIMESTAMP;
+    }
+
+    /**
+     * Forces this engine to advance its max_unsafe_auto_id_timestamp marker to at least the given timestamp.
+     * The engine will disable optimization for all append-only whose timestamp at most {@code newTimestamp}.
+     */
+    public abstract void forceUpdateMaxUnsafeAutoIdTimestamp(long newTimestamp);
```

Comment:
force suggests to me that whatever value you give this method will be set as the new value. It seems that the intended semantics are different. Maybe just call this updateMaxUnsafe...